kite-tools-parent/kite-tools/src/main/java/org/kitesdk/cli/commands/TarImportCommand.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2014 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kitesdk.cli.commands;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.hadoop.io.IOUtils;
+import org.kitesdk.cli.commands.tarimport.avro.TarFileEntry;
+import org.kitesdk.data.*;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+@Parameters(commandDescription = "Import files in tarball into a Dataset")
+public class TarImportCommand extends BaseDatasetCommand {
+
+  protected enum CompressionType {
+    NONE,
+    GZIP,
+    BZIP2,
+    XZ
+  }
+
+  private static final List<String> SUPPORTED_TAR_COMPRESSION_TYPES =
+      Lists.newArrayList("", "none", "gzip", "bzip2");
+
+  private static final long DEFAULT_BLOCK_SIZE = 128 * 1024 * 1024;
+
+  public TarImportCommand(Logger console) {
+    super(console);
+  }
+
+  @Parameter(description = "<tar path> <dataset URI>")
+  List<String> targets;
+
+  @Parameter(names = "--compression",
+      description = "Override compression type (none, gzip, bzip2)")
+  String compressionType = "";
+
+  @Override
+  public int run() throws IOException {
+    Preconditions.checkArgument(targets != null && targets.size() == 2,
+        "Tar path and target dataset URI are required.");
+
+    Preconditions.checkArgument(
+        SUPPORTED_TAR_COMPRESSION_TYPES.contains(compressionType),
+        "Compression type " + compressionType + " is not supported");
+
+    String source = targets.get(0);
+    String datasetUri = targets.get(1);
+
+    long blockSize = getConf().getLong("dfs.blocksize", DEFAULT_BLOCK_SIZE);
+
+    int success = 0;
+
+    View<TarFileEntry> targetDataset;
+    if (Datasets.exists(datasetUri)) {
+      console.info("Using existing dataset: {}", datasetUri);
```

Comment:
Nit: I think this one should be debug because this is what the user instructed the command to do. The unexpected case is when the dataset must be created.